### DISCUSSION TITLE
Target is a entity

### DISCUSSION THREAD

**Discussion by alanruttenberg** [CommonCoreOntology/CommonCoreOntologies#360] [Announcements]
http://www.ontologyrepository.com/CommonCoreOntologies/Target has no other parent than entity. The definition suggests it's a role

**Comment by harefb** [comment]
We concur with Alan. In the defense intelligence core ontology,  we treat target as a role that someone designates an entity of interest.

**Comment by rorudn** [comment]
This has been updated, I thought it had been for a while, but please recheck. Target is a defined class, subclass of Material Entity that is the object of some act of targeting.

**Comment by alanruttenberg** [comment]
Basing it on some act of targeting is not ideal in my opinion, certainly as
the definition of that term bears no formal relation to roles. "An
Intentional Act wherein some Agent maintains one or more entities as the
objects of potential action given some operational requirement."

The initial designation of something as a target is how the role comes into
existence. The definition suggests that the role goes out of existence when
no agent ''maintains" the entity as an object of potential action. What it
means for an agent to "maintains" is not clear to me. For instance, if a
target is designated and then filed away in a database, do you consider
there to be an agent that maintains it as an object of potential action?
What if everybody forgets about it until the appropriate database search
pulls it up.

It seems to me that BFO is clear on this. There's a  role, that of being a
target (perhaps there are specializations as well). That role is realized
when some action is taken that involves the bearer.  It would be a
reasonable specialization strategy to differentiate subclasses by the type
of action.

It's perfectly reasonable to do as you've done and create a defined class
of things that satisfy a criterion but I think the criteria is that the
thing bears the target role and that care needs to be made in
naming/documenting so that role and defined class of role bearer are clear.
My preference would be to have "target role" and define "target" in terms
of that. Following the strategy of naming information bearers in favor of
information entities, I could also see a different choice, something like
"target" for the role, and "target material bearer" for the defined class
(slightly tongue in cheek).

I'm guessing there may be different opinions about this, which would be
supported by allowing for labels that are associated with different
communities, something I've been advocating for.

Alan

**Comment by rorudn** [comment]
It's useful for me to quote the BFO elucidation of roles here though i know you are more than well acquainted.
"b is a role means: b is a realizable entity & b exists because there is some single bearer that is in **some special physical, social, or institutional set of circumstances** in which this bearer does not have to be& b is not such that, if it ceases to exist, then the physical make-up of the bearer is thereby changed." (emphasis mine)

So the problem is that I don't think that target is a role in this sense as there is no special physical, social, or institutional set of circumstances in which the bearer finds itself. A tin can placed on a picket fence is a target simply by the sharpshooter's intent to hit it with her next practice shot. There are myriad things of this kind (e.g. cargo, equipment, witness, bystander, rioter, assailant, New Yorker) These are just words that we use to designate a thing's manner of participation in a action. There is no New Yorker role, there's just the fact that the person resides in NYC. To turn these sorts of participations into roles trivializes the concept. Supreme Court Justice, yes that's a role, a refrigerator in a box in a container on a freighter is just a refrigerator in an act of transport that we like to call cargo. 

The question you pose about the record in the database doesn't have all the information I'd like to have to answer it with certainty on my part, but if the act of targeting is still in effect, then yes. If a person is placed on the FBI's most wanted list, then they are a high value target until no longer on that list, it doesn't matter whether anyone remembers that they are. Is the tin can still a target if the sharp shooter is distracted before taking the shot and starts doing something else? no.

I continue to plan to introduce an sibling to role and call it something like "Activity-based Role" to better capture these kinds. But until then, i use roles like everyone else, just a bit more grudgingly.

If you have a definition for target that would satisfy the some special physical, social, or institutional set of circumstances clause for all types of targets I'm keen to hear it.

**Comment by alanruttenberg** [comment]
I asked Barry about this. His response: "The infantry lives off target
lists. The same object may be on the target list today, not on the target
list tomorrow. Roles are introduced into BFO precisely to capture the sorts
of temporally modified is-a relations that would otherwise result. And
safeguard simple inheritance".

Regarding the definition, I think it can be reasonably argued that the
bearer is in, at least, a special social circumstance. It is perhaps not
clear enough what a social circumstance is, but a plausible interpretation
is that a social circumstance arises when there is involvement of a member
of society, even a society of one, with anything. In that case both are
involved in the social circumstance.

For a while I considered that there should be something more like an
ascription, borne by, in this case, the targeter, for much the same reasons
you cite. While I think there may be a need for something like this, I now
understand the BFO strategy here - attachment of the dependent as closely
as possible to the entity in focus. Another way I've heard roles described
is as "externally grounded".

Alan

Alan

**Comment by harefb** [comment]
The DICO definition for TARGET ROLE will be something like this:

A Role that inheres in an Agent, or other material entity, by virtue of that entity having been considered by some appropriate authority for possible engagement, or other action, because of the function the entity performs.

I would suggest this satisfies the “some special physical, social, or institutional set of circumstances “ clause requirement. A conflict, or potential conflict, is a social circumstance between a set of institutions.

We will include this item in a list we are assembling currently.

Regards,
Forrest



Forrest B. Hare, PhD, CISSP
SAIC Fellow
Solution Developer | Cyberspace Operations
571-419-0084 | forrest.b.hare@SAIC.com<mailto:forrest.b.hare@SAIC.com>
saic.com<http://www.saic.com/> |@SAICinc<https://twitter.com/SAICinc>
SAIC Redefining Ingenuity ™

**Comment by alanruttenberg** [comment]
On Fri, Oct 16, 2020 at 8:42 AM harefb <notifications@github.com> wrote:

> The DICO definition for TARGET ROLE will be something like this:
>
> A Role that inheres in an Agent, or other material entity, by virtue of
> that entity having been considered by some appropriate authority for
> possible engagement, or other action, because of the function the entity
> performs.
>
Hi Forrest,

Comments:

1. Since all agents (caveat, I'm not fond of the term agent - separate
conversation) are material entities, the inclusion of the agent clause is
superfluous.
2. The inclusion of the authority clause may belong in a subclass of
target, something like "authorized target".  A person can target something
just for fun, and a member of the armed forces may (improperly)  designate
a target without suitable authority.
3. The mention of function is slightly problematic for the following
reasons: Barry has always maintained that functions do not inhere in
organisms. We've argued about this some, with a case in point being ant or
bee specialization. Generally, however, where we might gravitate to say
function when referring to an organism the BFO interpretation is that we're
referring to a role.

I may be that the non-authorized case is not of relevance to your work,
though I think it plausibly might be. However if that's the case I'd
consider naming the term more specifically, at least as a synonym.
Within OBO there's a practice of having an "editor preferred label" which
has the property of being unique across the whole of the ontology and
typically more verbose in an effort to reduce misinterpretation. That may
be a practice worth considering - there's no implication that such a label
is commonly used. It's a term aimed at ontology editors.

These comments about the definition are not intended to interfere with your
need to cite authoritative definitions. However in the case where we
support multiple definitions of the same class, another definition might
address these issues.

--

I was also going to unpack what Barry said about the time-dependent is-a.
What he means is that since whether something is a target or not can
change, you can't have subclass assertions like: a airfield located in xxx
is a target, because the is-a is a-temporal (at all times). In practice
this sort of assertion can be avoided if you only assert target as the type
of an instance, since an entity can instantiate different classes at
different times.


> I would suggest this satisfies the “some special physical, social, or
> institutional set of circumstances “ clause requirement. A conflict, or
> potential conflict, is a social circumstance between a set of institutions.
>
> We will include this item in a list we are assembling currently.
>
> Regards,
> Forrest
>
>
>
> Forrest B. Hare, PhD, CISSP
> SAIC Fellow
> Solution Developer | Cyberspace Operations
> 571-419-0084 | forrest.b.hare@SAIC.com<mailto:forrest.b.hare@SAIC.com>
> saic.com<http://www.saic.com/> |@SAICinc<https://twitter.com/SAICinc>
> SAIC Redefining Ingenuity ™
>
>
>
> From: Alan Ruttenberg <notifications@github.com>
> Reply-To: CommonCoreOntology/CommonCoreOntologies <reply@reply.github.com>
>
> Date: Friday, October 16, 2020 at 08:36
> To: CommonCoreOntology/CommonCoreOntologies <
> CommonCoreOntologies@noreply.github.com>
> Cc: "Hare, Forrest B." <Forrest.B.Hare@saic.com>, Comment <
> comment@noreply.github.com>
> Subject: Re: [CommonCoreOntology/CommonCoreOntologies] Target is a entity
> (#84)
>
>
> EXTERNAL EMAIL -- This message originates from outside of SAIC
> I asked Barry about this. His response: "The infantry lives off target
> lists. The same object may be on the target list today, not on the target
> list tomorrow. Roles are introduced into BFO precisely to capture the
> sorts
> of temporally modified is-a relations that would otherwise result. And
> safeguard simple inheritance".
>
> Regarding the definition, I think it can be reasonably argued that the
> bearer is in, at least, a special social circumstance. It is perhaps not
> clear enough what a social circumstance is, but a plausible interpretation
> is that a social circumstance arises when there is involvement of a member
> of society, even a society of one, with anything. In that case both are
> involved in the social circumstance.
>
> For a while I considered that there should be something more like an
> ascription, borne by, in this case, the targeter, for much the same
> reasons
> you cite. While I think there may be a need for something like this, I now
> understand the BFO strategy here - attachment of the dependent as closely
> as possible to the entity in focus. Another way I've heard roles described
> is as "externally grounded".
>
> Alan
>
> Alan
>

**Comment by rorudn** [comment]
The inclusion of the entity in the target list is the act of targeting (or at least the output of it), which is the source of the external grounding. The swearing in of the Supreme Court Justice is another example of an action that binds a role to a carrier. The CCO method is consistent with the BFO method but avoids the need for the use of the temporal based object properties.

**Comment by alanruttenberg** [comment]
The use of roles does not necessarily require temporalized relations. The
pattern is that a process is defined in terms of which roles are realized,
and with the roles the bearers are implied (or may be asserted explicitly).
FWIW I don't think one can avoid use of temporalized relations. Time and
temporalized relations are an essential part of the BFO standard.
Alan

**Comment by alanruttenberg** [comment]
Illustration of the pattern. No temporal relations in sight.

[role realization.pdf](https://github.com/CommonCoreOntology/CommonCoreOntologies/files/5392885/role.realization.pdf)

**Comment by rorudn** [comment]
This representation of the event gives me pause.  "Tooth to be restored role"? I'm sorry but I can see this only as a contortion of reality to fit a model (BFO).  
I'm familiar with the realized in relation and yes, it doesn't need the temporal relations, but if I want to answer the question of how long RBG held the role of US Supreme Court Justice is that still the case?

**Comment by alanruttenberg** [comment]
On Fri, Oct 16, 2020 at 4:33 PM rorudn <notifications@github.com> wrote:

> This representation of the event gives me pause. "Tooth to be restored
> role"? I'm sorry but I can see this only as a contortion of reality to fit
> a model (BFO).
>
There needs to be a way to say what the various participants in a process
did. You need to distinguish the case where a doctor is doing the doctoring
and a doctor is being the patient. You can either make up some way to do
that, or you can express it with what's available in BFO. This is the way
to say it in BFO. There's nothing untowards about it - it's accurate,
definition conforming, and consistent. It's my preference to always use
something we know how to do over making stuff up.

> I'm familiar with the realized in relation and yes, it doesn't need the
> temporal relations, but if I want to answer the question of how long RBG
> held the role of US Supreme Court Justice is that still the case?
>
If you want to talk about when, you need to talk about when. It's the same
question as how long a certain site is to be considered a priority target.
BFO FOL allows this to be expressed clearly. OWL doesn't have the
expressivity to reason with ternary relationships. So we give the best we
can (currently) think of to get some reasoning mileage. Two routes are to
1) use temporalized relations 2) use the temporal projection relation from
a process to a temporal region, and then have an information artifact
that's about the temporal region. I've toyed with other ideas, but nothing
ready for prime time yet.

In the case of Ruth, we can talk about the process by which the role is
conferred and that in which it ceases, or we can talk about the process
during which she realizes the role.  Alternatively, we can skip the logic
altogether and use annotation properties. I'm in favor of the former, but
chacun a son gout. At some point we need to do some sort of hybrid
reasoning if we're going to go beyond OWL's capabilities to do temporal
reasoning. The purpose of the FOL is to, as best possible, give us a way to
check that the hybrid reasoning is correct. Even the FOL doesn't cover
everything. For instance it doesn't handle concrete domains so it can't
compute the duration of an interval. But there *are* a bunch of axioms
about how intervals work, and to the extent they are they provide some
ability to discriminate sense from nonsense.

**Comment by alanruttenberg** [comment]
> For instance it doesn't handle concrete domains so it can't compute the
duration of an interval. But there *are* a bunch of axioms about how
intervals work, and to the extent they are they provide some ability to
discriminate sense from nonsense.

I wanted to note that even though pure FOL can't do that, SMT
<https://en.wikipedia.org/wiki/Satisfiability_modulo_theories> theorem
provers can. On my (long) list of things I'd like to try is to extend BFO's
axiomatization to include numeric reasoning and then use an SMT prover like
Z3 <https://github.com/Z3Prover/z3> (also <https://rise4fun.com/z3>) to do
reasoning/prove consistency. One of the things I thought was very cool
about your CDF demo was that you used hybrid reasoning - adding graph
matching to OWL.

**Comment by rorudn** [comment]
> There needs to be a way to say what the various participants in a process
did. You need to distinguish the case where a doctor is doing the doctoring
and a doctor is being the patient. You can either make up some way to do
that, or you can express it with what's available in BFO. 

The doctor as doctor vs doctor as patient example uses dependents that I agree are roles. 

Let me change to the case of a rioter. I know what this participant in the process of a riot did in virtue of knowing that they were the agent in a process of rioting and that that process was part of the riot process. I don't see an argument to support the claim that this is made up unless there's something amiss about specializing participation to agency. If I want to document what a witness did in the riot, then I record that they are the agent in an act of observation and that this act is part of the riot. 

There would be some roles that would be realized in this riot, for example police officers, reporters, emergency medical technicians, fire fighters, but to know what they did, the realizations would be in sub-processes. What I understand you to be saying is that every unique participation in a process is documented by creating a role. So in the case of the riot reporter, there would be a reporter narrating a riot role (to distinguish their performance from the performance of the camera person that films this narration).
  
To restate the original issue, you propose that target should be treated as a defined class in terms a target role instead of its current treatment as a defined class in terms of being the object of some act of targeting.  The two are not incompatible. The act of targeting would the process that creates the target role, its external grounding. Having this act and the act which results in the   loss of the role is,  as you point out, very useful for recording the temporal interval over which the role is borne. The act also generalizes the act of consideration for possible engagement in the definition offered by Forrest. In release 1.3 of CCO based upon interactions with the industrial ontology foundry we've added roles for Part, Component and System which are examples of classes that I originally held as being examples of the activity-based roles I mentioned earlier. I've been convinced that these are participations that are special enough to meet the BFO definition. Because of trivial cases of targeting where the target is determined by not much more than whimsy (tin can on fence) it was not added as a role. I think adding some sub-type of Target Role would be a good compromise. 

On the broader topic of using roles to document the manner of an object's participation in a process, I'm trying to hold a line for roles having a special circumstance. For me, special does not mean unique. Users of CCO can accomplish the same thing you are doing with roles by using specializations of the participation property and processes. So a tooth to be extracted role would be equivalent to a tooth that is the object of an act of tooth extraction.

**Comment by harefb** [comment]
I’m honestly not sure I understand all the below, but I did key in on one point-

I agree that the two views of target as being a role and being the “recipient in a process” (to paraphrase) are not incompatible.

In fact, I was writing a sample triple just yesterday and that appeared to be an elegant solution to modeling transitive actions (not properties) with BFO as they pertain to military conflict.

When one tank fires on another tank, it doesn’t just participate in a firefight.  It is doing something to something else (the recipient of the action) and that point needs to be explicit.  Conveniently, the recipient is the object that has been assigned the role of target.  In plain language- the first tank fires a round at the target tank.

Whether that designation of target was instantaneous on the battlefield or the result of a process that happened at HQ the day before, it still happened before the action.  I still think the primary view of the concept is the role.  It is just that the role is one of being designated the recipient of an action.

Baby steps….

-Forrest



Forrest B. Hare, PhD, CISSP
SAIC Fellow
Solution Developer | Cyberspace Operations
571-419-0084 | forrest.b.hare@SAIC.com<mailto:forrest.b.hare@SAIC.com>
saic.com<http://www.saic.com/> |@SAICinc<https://twitter.com/SAICinc>
SAIC Redefining Ingenuity ™

**Comment by alanruttenberg** [comment]
The term rioter is a social term if I've ever heard one. It's not at all clear that the alleged rioter has anything in their mind regarding rioting. They may view themselves as protester, or reacting in defense. The role of rioter is externally grounded - it's a role that is ascribed to a person from a particular political perspective. 

What is the sound of one hand clapping? How do you separate what the EMTs are doing from who they are doing it to?
Having a subprocess is a perfectly reasonable thing to have, but in my view it's a funny kind of process to talk about what an EMT is doing without them doing it to someone, just like it is hard to imagine a process of one hand clapping. 

Moreover that someone is an EMT is something that is known in advance, and certain acts in such a situation are expected by virtue of their license. Realizables are sometimes call "future pointing". That's the implication of "of a type some instances of which are realized in processes of a correlated type".

Not everything the individual person that is an EMT does or participates in a realization of their role. It's not a realization of an EMT role for them to have a heart attack on the job, to be shot, or to talk about last week's football game. But at some point the person who is an EMT starts having a patient role, and that patient role also points to a certain set of correlated processes that are realizations of that patient role. 

Regarding tin can on a fence, there's whimsical shooting - I'm shooting in the general direct of the barn hoping I'll hit something. But in the case there being a conscious act of targeting the tin can we do have a special circumstance - the intention, the proximity, that there will not be sanction for shooting the tin can.  If the person is distracted by a conversation, they may turn back again to their task and shoot at the same thing they planned to. So the role pre-exists and can outlive other intervening processes. It's future pointing.

The criterion for being a role is not that the circumstances are "special enough" What does that even mean? How could different people come to the same judgement about whether there's a role, a requirements if their is to be effective data exchange, if each one has to make an ill-defined "special enough" judgement. 

>Realizable entity: ELUCIDATION: b is a realizable entity means: b is a specifically dependent continuant that inheres in some independent continuant which is not a spatial region and is of a type some instances of which are realized in processes of a correlated type [058-BFO] 

>Role: ELUCIDATION: b is a role means: b is a realizable entity & b exists because there is some single bearer that is in some special physical, social, or institutional set of circumstances in which this bearer does not have to be & b is not such that, if it ceases to exist, then the physical make-up of the bearer is thereby changed [061-BFO]

A role needs to satisfy both these things. While the elucidation of role does say "special physical, social, or institutional set of circumstances", you are giving too much weight to "special" and not enough weight to "of a type some instances of which are realized in processes of a correlated type". It's clear you are putting more weight on "special" than appropriate as the person who wrote that definition, Barry, has clearly stated that something like target is precisely the sort of thing he intended. I think "Special" should be understood in the same sort of way as "in which this bearer does not have to be".  "Special" means it's not any old situation. It's the kind that is correlated with the realization. Examples of EMT realizations and not-realizations above are relevant.

Other than that, it's hard to see any useful content in the sentence. What situation *isn't* social or institutional or physical?  And "special", alone, isn't the sort of thing that would be at all useful for the intended purpose of BFO, which is to help set standards that help interoperability, something that in turn depends on distributed/uncoordinated people being able to come to the same conclusion about what the nature of a thing is. 

Here is a better decision tree to decide whether something is a role. I'll use "entity" to mean IC in question. 

1. Is there something about that entity that predicts that it will be involved in certain ways in certain circumstances?
2. Is that something an essential characteristic of the entity? 
3. Is it possible that the something goes away without the bearer's physical constitution changing?

If the answers are 1: Yes : No 3: Yes. then the something is probably a role.
Answers for target: Yes, No, Yes.

I do think there's a fuzzy area when something relatively spontaneous happens. For instance at the bar someone take a
swing at someone else. There's a distinction between the who does what, and there's even the a basic question of who the
"aggressor" is, assuming that's the role in question. There's also the epistemology question - suppose it takes some
time to determine who started the fight, and so one learns who the aggressor is only after the fact. I know how Werner
would answer - that it doesn't matter *when* you find out. There's a parallel in disease/diagnosis.

On the other hand seeming spontaneity doesn't necessarily rule out realizables. It may spontaneously be the case that a
wind comes along and nudges a rock, which subsequently rolls down a hill. In that case the rock is realizing it's
disposition, related to having mass, be attracted to a body with higher mass.

But suppose we don't use roles in the spontaneously social case. The usual fallback is to use subproperties of
participation. However, usually we advise using as few properties as possible for engineering reasons. For instance, there's less logical expressivity around properties than classes (at least in first-order languages). But perhaps it's appropriate in some cases.

**Comment by oliviahobai** [comment]
Moved to discussion to continue development on what it means to be a target.
